network/p2p/readSubscription.go

```diff
@@ -79,26 +79,17 @@ func (r *readSubscription) receiveLoop(wg *sync.WaitGroup) {
 			return
 		}
 
-		pid, err := messageSigningID(rawMsg)
-		if err != nil {
-			r.log.Err(err).Msg("failed to validate peer ID of incoming message")
-			return
+		validatorData, ok := rawMsg.ValidatorData.(validator.ValidatorData)
+		if !ok {
+			r.log.Error().Str("raw_msg", rawMsg.String()).Msg("validator data missing")
```

Comment:
Now (see [prior discussion](https://github.com/onflow/flow-go/pull/1289#discussion_r711585971)) my issue is this is logging and continuing, i.e. swallowing the error. This situation is probably the sign of a bug (and the error should say so), can't guarantee message validation, and should not lead to the message being processed. I'd perform a return here, making sure we do not reach the callback.

Reply:
Oops, that's what I meant to do here

Reply:
updated, thanks for catching this